doc/14_data_frames.md

```diff
@@ -204,11 +204,52 @@ spark.sql("SELECT Count(*) FROM mycatalog.ks.tab WHERE key = 1").explain
 
 #### Direct Join
 
-Joins with a Cassandra Table using a Partition Key as an equals will be automatically
-converted into a `joinWithCassandraTable` style join if it is more efficent to 
-query Cassandra that way.
+Joins with a Cassandra Table using a Partition Key may be automatically
+converted into a `joinWithCassandraTable` style join if it is more efficient to 
+query Cassandra that way. 
 
-//TODO add full docs on configuration
+By default (`directJoinSetting=auto`) the Spark Cassandra Connector converts a 
+join to a Direct Join when the following formula is true:
+
+    (table size * directJoinSizeRatio) > size of keys
+
+`directJoinSizeRatio` is a setting thay my be adjusted just like any other
+Spark Cassandra Setting. See [Parameters section](reference.md#cassandra-connection-parameters)
+for details.
+
+Automatic Direct Join conversion may be permanently disabled or enabled with 
+`directJoinSetting=off` and `directJoinSetting=on` settings.
+
+For example to disregard `directJoinSizeRatio` parameter and convert all the 
+suitable joins to Diract Joins start `spark-sql` with:
+
+    spark-sql --conf directJoinSetting=on
+    
+Note that not all joins are suitable from Direct Join conversion. The following 
+conditions must be met for the conversion to happen:
+ - At least one side of the join is a CassandraSourceRelation
+ - The join condition fully restricts the partition key
+ - The join keys types must match
+
+Direct Join example with the following the table:
```

Comment:
I'm not exactly sure what you're intending to say here but maybe you just need to remove the last "the" for the sentence to make sense.